UserService/UserService.API/Middlewares/ExceptionHandlingMiddleware.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Net;
+using System.Text.Json;
+using UserService.BLL.Exceptions;
+
+namespace UserService.API.Middlewares
+{
+    public class ExceptionHandlingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger<ExceptionHandlingMiddleware> _logger;
+
+        public ExceptionHandlingMiddleware(
+            RequestDelegate next,
+            ILogger<ExceptionHandlingMiddleware> logger)
+        {
+            _next = next;
+            _logger = logger;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            try
+            {
+                await _next(context);
+            }
+            catch (NotFoundException exception)
+            {
+                context.Response.StatusCode = StatusCodes.Status404NotFound;
+                await HandleExceptionAsync(context, exception);
+            }
+            catch (BadRequestException exception)
+            {
+                context.Response.StatusCode = StatusCodes.Status400BadRequest;
+                await HandleExceptionAsync(context, exception);
+            }
+            catch (UnauthorizedException exception)
+            {
+                context.Response.StatusCode = StatusCodes.Status401Unauthorized;
+                await HandleExceptionAsync(context, exception);
+            }
+            catch (Exception exception)
+            {
+                context.Response.StatusCode = StatusCodes.Status500InternalServerError;
+                await HandleExceptionAsync(context, exception);
+            }
```

Comment:
you can pass status code as a parameter to HandleExceptionAsync method and set it to context.Response.StatusCode there